core-bundle/src/Security/Authentication/FrontendPreviewAuthenticator.php

```diff
@@ -48,23 +53,19 @@ public function authenticateFrontendUser(string $username, bool $showUnpublished
 
         $token = new UsernamePasswordToken($user, 'contao_frontend', $user->getRoles());
 
-        try {
-            $session = $this->requestStack->getSession();
-        } catch (SessionNotFoundException) {
+        if (!$session = $this->getSession()) {
             return false;
         }
 
-        $session->set('_security_contao_frontend', serialize($token));
+        $this->updateToken($token);
```

Comment:
This will never be executed if there is no session, but `$this->tokenStorage->setToken($token)` could also be run without a session. Is this logic still correct?